db/lib/database-access.js

```diff
@@ -0,0 +1,101 @@
+/** @import { BlockEvent } from '../../backend/lib/rpc-service/data-types.js' */
+/** @import { Static } from '@sinclair/typebox' */
+/** @import {Queryable} from '@filecoin-station/deal-observer-db' */
+import { ActiveDealDbEntry } from '@filecoin-station/deal-observer-db/lib/types.js'
```

Comment:
This creates a circular reference between `db` and `backend`. Circular references are almost always a bad thing.

Please move the typedefs (model definitions) to `db/lib` too.

Reply:
It does [live](https://github.com/filecoin-station/deal-observer/blob/main/db/lib/types.js#L14) in `db/lib`.

Reply:
Oh, I see. In that case use a relative path in the import statement.

```suggestion
import { ActiveDealDbEntry } from './types.js'
```